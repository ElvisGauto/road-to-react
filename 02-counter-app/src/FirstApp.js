import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const FirstApp = ( { greeting, subtitle } ) => {
    return (        
        <>
            <h1>{ greeting }</h1>
            <p>{ subtitle }</p>
        </>
    );
}

FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'Soy un subtitle'
}

export default FirstApp;