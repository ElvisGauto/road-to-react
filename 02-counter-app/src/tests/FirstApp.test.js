import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { render }  from '@testing-library/react';
import FirstApp from '../FirstApp';

describe('<FirstApp /> tests', () => {
    // test('should show message "Hola, soy Goku"', async () => {
    //     const greeting = 'Hola, soy Goku';

    //     const { getByText } = render(<FirstApp greeting={greeting} />);

    //     expect( getByText(greeting) ).toBeInTheDocument();
    // })
    test('should show <FirstApp /> correctly', () => {
        const greeting = 'Hola, soy Goku';
        const wrapper = shallow(<FirstApp greeting={greeting} />);

        expect( wrapper ).toMatchSnapshot();
    })
    test('should show subtitile send by props', () => {
        const greeting = 'Hola, soy Goku';
        const subtitle = 'Soy un subtitle';

        const wrapper = shallow(
            <FirstApp 
                greeting={greeting}
                subtitle={subtitle} 
            />
        );
        
        const paragraph = wrapper.find('p').text();

        expect( paragraph ).toBe( subtitle );
        
    })

});