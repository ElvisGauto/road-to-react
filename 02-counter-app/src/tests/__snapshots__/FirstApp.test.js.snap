// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstApp /> tests should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitle
  </p>
</Fragment>
`;
